contracts/utils/VaultQueue.sol

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {
+    Initializable
+} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault, IDepositContract} from "../interfaces/IRibbon.sol";
+import {ICRV} from "../interfaces/ICRV.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {DSMath} from "../vendor/DSMath.sol";
+
+contract VaultQueue is
+    Initializable,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    enum TransferAction {INTERVAULT, WITHDRAW}
```

Comment:
Thinking about it, why do we need `TransferAction`? Because if the `srcVault` and `dstVault` are the same, that implies that it is an `INTERVAULT` transfer, so it feels redundant

Reply:
I was planning to add `SKIP_ROUND` and different validation checks for each `TransferAction`